diff -Naur A/src/engine/engine_util_errmem.c B/src/engine/engine_util_errmem.c
--- A/src/engine/engine_util_errmem.c	2023-03-20 16:33:36.000000000 +0100
+++ B/src/engine/engine_util_errmem.c	2024-10-25 12:01:50.352196121 +0200
@@ -106,7 +106,7 @@
     // get time
     time(&rawtime);
 
-#if defined(_POSIX_C_SOURCE) || defined(__APPLE__) || defined(__STDC_VERSION_TIME_H__)
+#if defined(_POSIX_C_SOURCE) || defined(__APPLE__) || defined(__EMSCRIPTEN__) || defined(__STDC_VERSION_TIME_H__)
     localtime_r(&rawtime, &timeinfo);
 #elif _MSC_VER
     localtime_s(&timeinfo, &rawtime);