diff --git a/cmake/MujocoDependencies.cmake b/cmake/MujocoDependencies.cmake
--- a/cmake/MujocoDependencies.cmake
+++ b/cmake/MujocoDependencies.cmake
@@ -207,34 +207,34 @@ target_link_options(SdfLib PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
 
 set(ENABLE_DOUBLE_PRECISION ON)
 set(CCD_HIDE_ALL_SYMBOLS ON)
-findorfetch(
-  USE_SYSTEM_PACKAGE
-  OFF
-  PACKAGE_NAME
-  ccd
-  LIBRARY_NAME
-  ccd
-  GIT_REPO
-  https://github.com/danfis/libccd.git
-  GIT_TAG
-  ${MUJOCO_DEP_VERSION_ccd}
-  TARGETS
-  ccd
-  EXCLUDE_FROM_ALL
-)
-target_compile_options(ccd PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
-target_link_options(ccd PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
+# findorfetch(
+#   USE_SYSTEM_PACKAGE
+#   OFF
+#   PACKAGE_NAME
+#   ccd
+#   LIBRARY_NAME
+#   ccd
+#   GIT_REPO
+#   https://github.com/danfis/libccd.git
+#   GIT_TAG
+#   ${MUJOCO_DEP_VERSION_ccd}
+#   TARGETS
+#   ccd
+#   EXCLUDE_FROM_ALL
+# )
+# target_compile_options(ccd PRIVATE ${MUJOCO_MACOS_COMPILE_OPTIONS})
+# target_link_options(ccd PRIVATE ${MUJOCO_MACOS_LINK_OPTIONS})
 
 # libCCD has an unconditional `#define _CRT_SECURE_NO_WARNINGS` on Windows.
 # TODO(stunya): Remove this after https://github.com/danfis/libccd/pull/77 is merged.
-if(WIN32)
-  if(MSVC)
-    # C4005 is the MSVC equivalent of -Wmacro-redefined.
-    target_compile_options(ccd PRIVATE /wd4005)
-  else()
-    target_compile_options(ccd PRIVATE -Wno-macro-redefined)
-  endif()
-endif()
+# if(WIN32)
+#   if(MSVC)
+#     # C4005 is the MSVC equivalent of -Wmacro-redefined.
+#     target_compile_options(ccd PRIVATE /wd4005)
+#   else()
+#     target_compile_options(ccd PRIVATE -Wno-macro-redefined)
+#   endif()
+# endif()
 
 if(MUJOCO_BUILD_TESTS)
   set(ABSL_PROPAGATE_CXX_STD ON)
